pkg/k8s/observedstate.go

```diff
@@ -0,0 +1,22 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8s
+
+func HasComputedFieldsUnderObservedState(kind, version string) bool {
```

Comment:
Nit: This method needs a comment.  I think "Computed Field" is a Terraform thing, maybe this should  "OutputFieldsAreUnderObservedState" or maybe just "GetOutputField(u *unstructured.Unstructured, fieldPath string)"?

Reply:
Changed the function name to `OutputOnlyFieldsAreUnderObservedState`. It needs to take the smallest set of params because it is also called by CRD generation code (which doesn't have unstructured).